pkg/platform/common/vmlayer/vmprovider.go

```diff
@@ -408,38 +400,30 @@ func (v *VMPlatform) InitCommon(ctx context.Context, platformConfig *platform.Pl
 	}
 	v.HAManager = haMgr
 
-	if !platformConfig.TestMode {
-		err := v.VMProperties.CommonPf.InitCloudletSSHKeys(ctx, platformConfig.AccessApi)
-		if err != nil {
-			return err
-		}
-		go v.VMProperties.CommonPf.RefreshCloudletSSHKeys(platformConfig.AccessApi)
-	}
-
 	var err error
 	if err = v.InitProps(ctx, platformConfig); err != nil {
 		return err
 	}
-	v.initDebug(v.VMProperties.CommonPf.PlatformConfig.NodeMgr)
-
+	if platformConfig.CrmOnEdge {
+		v.initDebug(v.VMProperties.CommonPf.PlatformConfig.NodeMgr)
+	}
 	v.VMProvider.InitData(ctx, caches)
 
 	updateCallback(edgeproto.UpdateTask, "Fetching API access credentials")
 	if err = v.VMProvider.InitApiAccessProperties(ctx, platformConfig.AccessApi, platformConfig.EnvVars); err != nil {
 		return err
 	}
-	v.FlavorList, err = v.VMProvider.GetFlavorList(ctx)
-	if err != nil {
-		log.SpanLog(ctx, log.DebugLevelInfra, "GetFlavorList failed", "err", err)
-		return err
-	}
 	var cloudlet edgeproto.Cloudlet
 	if !v.Caches.CloudletCache.Get(v.VMProperties.CommonPf.PlatformConfig.CloudletKey, &cloudlet) {
 		return fmt.Errorf("unable to find cloudlet key in cache")
 	}
+	// TODO: resolve why there are two ways to store/use the external network name
 	v.VMProperties.PlatformExternalNetwork = cloudlet.InfraConfig.ExternalNetworkName
+	if cloudlet.InfraConfig.ExternalNetworkName != "" {
```

Comment:
This change essentially fixes https://github.com/edgexr/edge-cloud-platform/issues/337 since now infraconfig takes precedence.

Reply:
It's still confusing to me why we have two ways (env var and cloudlet.infraconfig) of specifying the same thing. But if I helped at least address some issue here, that's great :)

Reply:
I think this changes the behavior, but probably we should retire one way altogether.